### PR TITLE
Create com.github.bilelmoussaoui.Authenticator.json

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
A flatpak pacakge of Authenticator 
https://github.com/bilelmoussaoui/Authenticator/

### CODE REVIEW

**Comment by bilelmoussaoui** [comment]
Not sure about the finish-args for the database file.

File: com.github.bilelmoussaoui.Authenticator.json [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+   "runtime-version": "3.26",
+   "sdk": "org.gnome.Sdk",
+   "command": "authenticator",
+   "finish-args": [
+     /* X11 + XShm */
+     "--share=ipc", "--socket=x11",
+     /* Wayland */
+     "--socket=wayland",
+     /* Filesystem */
+     "--filesystem=~/.config/Authenticator:ro",
```
Comment (TingPing) [inline com.github.bilelmoussaoui.Authenticator.json]: The application should respect XDG and this isn't needed.

File: com.github.bilelmoussaoui.Authenticator.json [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+         "type": "archive",
+         "url": "http://downloads.sourceforge.net/project/zbar/zbar/0.10/zbar-0.10.tar.bz2",
+         "sha256": "234efb39dbbe5cef4189cc76f37afbe3cfcfb45ae52493bfe8e191318bdbadc6"
+       }]
+     },
+     {
+       "name": "pyotp",
+       "buildsystem": "simple",
+       "build-commands": [
+         "python3 setup.py install --prefix=/app --root=/"
```
Comment (TingPing) [inline com.github.bilelmoussaoui.Authenticator.json]: `--root=/` should be redundant for all of these.

File: com.github.bilelmoussaoui.Authenticator.json [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+         "url": "https://github.com/NaturalHistoryMuseum/pyzbar/archive/v0.1.4.tar.gz",
+         "sha256": "7387899656e4e9564b7e507b2c26729c4863b7a79bbaf29c80c956a5654d2ee0"
+       }]
+     },
+     {
+       "name": "gnome-screenshot",
+       "sources": [{
+         "type": "git",
+         "url": "https://git.gnome.org/browse/gnome-screenshot",
+         "branch": "gnome-3-22"
```
Comment (TingPing) [inline com.github.bilelmoussaoui.Authenticator.json]: This should be a versioned release.

**Comment by TingPing** [comment]
bot, build com.github.bilelmoussaoui.Authenticator

File: com.github.bilelmoussaoui.Authenticator.json [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+   ],
+   "build-options": {
+     "cflags": "-O2 -g",
+     "cxxflags": "-O2 -g",
+     "env": {
+       "V": "1"
+     }
+   },
+   "modules": [{
+       "name": "zbar",
```
Comment (nedrichards) [inline com.github.bilelmoussaoui.Authenticator.json]: this needs a reconfigure to fix aarch64

File: com.github.bilelmoussaoui.Authenticator.json (hunk 1 above)
Comment (bilelmoussaoui) [reply com.github.bilelmoussaoui.Authenticator.json]: Alright, currently there's no possible way to import a database/export it. As the tokens are stored using libsecret and I don't plan to change that anytime soon.

So in this case, I should just remove the filesystem finish-args?

File: com.github.bilelmoussaoui.Authenticator.json (hunk 4 above)
Comment (bilelmoussaoui) [reply com.github.bilelmoussaoui.Authenticator.json]: Have added `"rm-configure": true`. Is that enough?

File: com.github.bilelmoussaoui.Authenticator.json (hunk 3 above)
Comment (bilelmoussaoui) [reply com.github.bilelmoussaoui.Authenticator.json]: Fixed

File: com.github.bilelmoussaoui.Authenticator.json (hunk 2 above)
Comment (bilelmoussaoui) [reply com.github.bilelmoussaoui.Authenticator.json]: Fixed

File: com.github.bilelmoussaoui.Authenticator.json (hunk 4 above)
Comment (bilelmoussaoui) [reply com.github.bilelmoussaoui.Authenticator.json]: It doesn't really work with reconfigure on x86_64
```
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'config'.
libtoolize: linking file 'config/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'config'.
libtoolize: linking file 'config/libtool.m4'
libtoolize: linking file 'config/ltoptions.m4'
libtoolize: linking file 'config/ltsugar.m4'
libtoolize: linking file 'config/ltversion.m4'
libtoolize: linking file 'config/lt~obsolete.m4'
configure.ac:10: installing 'config/compile'
configure.ac:6: installing 'config/missing'
automake: warnings are treated as errors
gtk/Makefile.am.inc:13: warning: '%'-style pattern rules are a GNU make extension
Makefile.am:32:   'gtk/Makefile.am.inc' included from here
gtk/Makefile.am.inc:17: warning: '%'-style pattern rules are a GNU make extension
Makefile.am:32:   'gtk/Makefile.am.inc' included from here
pygtk/Makefile.am.inc:20: warning: '%'-style pattern rules are a GNU make extension
Makefile.am:35:   'pygtk/Makefile.am.inc' included from here
qt/Makefile.am.inc:11: warning: '%'-style pattern rules are a GNU make extension
Makefile.am:39:   'qt/Makefile.am.inc' included from here
qt/Makefile.am.inc:14: warning: '%'-style pattern rules are a GNU make extension
Makefile.am:39:   'qt/Makefile.am.inc' included from here
test/Makefile.am.inc:60: warning: '%'-style pattern rules are a GNU make extension
Makefile.am:45:   'test/Makefile.am.inc' included from here
Makefile.am:70: warning: '%'-style pattern rules are a GNU make extension
Makefile.am:73: warning: '%'-style pattern rules are a GNU make extension
/usr/share/automake-1.15/am/ltlibrary.am: warning: 'gtk/libzbargtk.la': linking libtool libraries using a non-POSIX
/usr/share/automake-1.15/am/ltlibrary.am: archiver requires 'AM_PROG_AR' in 'configure.ac'
Makefile.am:32:   'gtk/Makefile.am.inc' included from here
gtk/Makefile.am.inc:1:   while processing Libtool library 'gtk/libzbargtk.la'
/usr/share/automake-1.15/am/ltlibrary.am: warning: 'plugin/libzbarplugin.la': linking libtool libraries using a non-POSIX
/usr/share/automake-1.15/am/ltlibrary.am: archiver requires 'AM_PROG_AR' in 'configure.ac'
Makefile.am:43:   'plugin/Makefile.am.inc' included from here
plugin/Makefile.am.inc:1:   while processing Libtool library 'plugin/libzbarplugin.la'
/usr/share/automake-1.15/am/ltlibrary.am: warning: 'pygtk/zbarpygtk.la': linking libtool libraries using a non-POSIX
/usr/share/automake-1.15/am/ltlibrary.am: archiver requires 'AM_PROG_AR' in 'configure.ac'
Makefile.am:35:   'pygtk/Makefile.am.inc' included from here
pygtk/Makefile.am.inc:1:   while processing Libtool library 'pygtk/zbarpygtk.la'
/usr/share/automake-1.15/am/ltlibrary.am: warning: 'python/zbar.la': linking libtool libraries using a non-POSIX
/usr/share/automake-1.15/am/ltlibrary.am: archiver requires 'AM_PROG_AR' in 'configure.ac'
Makefile.am:29:   'python/Makefile.am.inc' included from here
python/Makefile.am.inc:1:   while processing Libtool library 'python/zbar.la'
/usr/share/automake-1.15/am/ltlibrary.am: warning: 'qt/libzbarqt.la': linking libtool libraries using a non-POSIX
/usr/share/automake-1.15/am/ltlibrary.am: archiver requires 'AM_PROG_AR' in 'configure.ac'
Makefile.am:39:   'qt/Makefile.am.inc' included from here
qt/Makefile.am.inc:1:   while processing Libtool library 'qt/libzbarqt.la'
/usr/share/automake-1.15/am/ltlibrary.am: warning: 'zbar/libzbar.la': linking libtool libraries using a non-POSIX
/usr/share/automake-1.15/am/ltlibrary.am: archiver requires 'AM_PROG_AR' in 'configure.ac'
Makefile.am:8:   while processing Libtool library 'zbar/libzbar.la'
Makefile.am: installing 'config/depcomp'
autoreconf: Entering directory `.'
autoreconf: configure.ac: not using Gettext
autoreconf: running: aclocal --force -I config
autoreconf: configure.ac: tracing
autoreconf: running: libtoolize --copy --force
libtoolize: putting auxiliary files in AC_CONFIG_AUX_DIR, 'config'.
libtoolize: copying file 'config/ltmain.sh'
libtoolize: putting macros in AC_CONFIG_MACRO_DIRS, 'config'.
libtoolize: copying file 'config/libtool.m4'
libtoolize: copying file 'config/ltoptions.m4'
libtoolize: copying file 'config/ltsugar.m4'
libtoolize: copying file 'config/ltversion.m4'
libtoolize: copying file 'config/lt~obsolete.m4'
autoreconf: running: /usr/bin/autoconf --force
autoreconf: running: /usr/bin/autoheader --force
autoreconf: running: automake --add-missing --copy --force-missing
configure.ac:10: installing 'config/compile'
configure.ac:6: installing 'config/missing'
automake: warnings are treated as errors
gtk/Makefile.am.inc:13: warning: '%'-style pattern rules are a GNU make extension
Makefile.am:32:   'gtk/Makefile.am.inc' included from here
gtk/Makefile.am.inc:17: warning: '%'-style pattern rules are a GNU make extension
Makefile.am:32:   'gtk/Makefile.am.inc' included from here
pygtk/Makefile.am.inc:20: warning: '%'-style pattern rules are a GNU make extension
Makefile.am:35:   'pygtk/Makefile.am.inc' included from here
qt/Makefile.am.inc:11: warning: '%'-style pattern rules are a GNU make extension
Makefile.am:39:   'qt/Makefile.am.inc' included from here
qt/Makefile.am.inc:14: warning: '%'-style pattern rules are a GNU make extension
Makefile.am:39:   'qt/Makefile.am.inc' included from here
test/Makefile.am.inc:60: warning: '%'-style pattern rules are a GNU make extension
Makefile.am:45:   'test/Makefile.am.inc' included from here
Makefile.am:70: warning: '%'-style pattern rules are a GNU make extension
Makefile.am:73: warning: '%'-style pattern rules are a GNU make extension
/usr/share/automake-1.15/am/ltlibrary.am: warning: 'gtk/libzbargtk.la': linking libtool libraries using a non-POSIX
/usr/share/automake-1.15/am/ltlibrary.am: archiver requires 'AM_PROG_AR' in 'configure.ac'
Makefile.am:32:   'gtk/Makefile.am.inc' included from here
gtk/Makefile.am.inc:1:   while processing Libtool library 'gtk/libzbargtk.la'
/usr/share/automake-1.15/am/ltlibrary.am: warning: 'plugin/libzbarplugin.la': linking libtool libraries using a non-POSIX
/usr/share/automake-1.15/am/ltlibrary.am: archiver requires 'AM_PROG_AR' in 'configure.ac'
Makefile.am:43:   'plugin/Makefile.am.inc' included from here
plugin/Makefile.am.inc:1:   while processing Libtool library 'plugin/libzbarplugin.la'
/usr/share/automake-1.15/am/ltlibrary.am: warning: 'pygtk/zbarpygtk.la': linking libtool libraries using a non-POSIX
/usr/share/automake-1.15/am/ltlibrary.am: archiver requires 'AM_PROG_AR' in 'configure.ac'
Makefile.am:35:   'pygtk/Makefile.am.inc' included from here
pygtk/Makefile.am.inc:1:   while processing Libtool library 'pygtk/zbarpygtk.la'
/usr/share/automake-1.15/am/ltlibrary.am: warning: 'python/zbar.la': linking libtool libraries using a non-POSIX
/usr/share/automake-1.15/am/ltlibrary.am: archiver requires 'AM_PROG_AR' in 'configure.ac'
Makefile.am:29:   'python/Makefile.am.inc' included from here
python/Makefile.am.inc:1:   while processing Libtool library 'python/zbar.la'
/usr/share/automake-1.15/am/ltlibrary.am: warning: 'qt/libzbarqt.la': linking libtool libraries using a non-POSIX
/usr/share/automake-1.15/am/ltlibrary.am: archiver requires 'AM_PROG_AR' in 'configure.ac'
Makefile.am:39:   'qt/Makefile.am.inc' included from here
qt/Makefile.am.inc:1:   while processing Libtool library 'qt/libzbarqt.la'
/usr/share/automake-1.15/am/ltlibrary.am: warning: 'zbar/libzbar.la': linking libtool libraries using a non-POSIX
/usr/share/automake-1.15/am/ltlibrary.am: archiver requires 'AM_PROG_AR' in 'configure.ac'
Makefile.am:8:   while processing Libtool library 'zbar/libzbar.la'
Makefile.am: installing 'config/depcomp'
autoreconf: automake failed with exit status: 1
Error: module zbar: module zbar: Child process exited with code 1
```

Comment (nedrichards) [reply com.github.bilelmoussaoui.Authenticator.json]: The pattern is usually that you need an autoconf.sh with autoreconf -vfi as per: https://github.com/flathub/fr.free.Homebank/blob/91c81b69cfe98bfea5593ba6475c3b69b7308d04/fr.free.Homebank.json#L58

**Comment by nedrichards** [comment]
bot, build com.github.bilelmoussaoui.Authenticator

File: com.github.bilelmoussaoui.Authenticator.json (hunk 4 above)
Comment (bilelmoussaoui) [reply com.github.bilelmoussaoui.Authenticator.json]: I have done that on my local machine and it doesn't work at all.. I'm getting the same error message. See the logs in my latest comment

Comment (bilelmoussaoui) [reply com.github.bilelmoussaoui.Authenticator.json]: Figured out how to fix that.

**Comment by nedrichards** [comment]
bot, build com.github.bilelmoussaoui.Authenticator

**Comment by bilelmoussaoui** [comment]
The build fails during the build of gnome-screenshot on aarch64
```
[0/16] Generating screenshot-resources_c with a custom command.
[0/16] Generating screenshot-resources_h with a custom command.
[0/16] Generating org.gnome.Screenshot.desktop_merge with a custom command.
[0/16] Generating org.gnome.Screenshot.metainfo.xml_merge with a custom command.
[1/16] Generating screenshot-resources_h with a custom command.
[1/16] Compiling C object 'src/gnome-screenshot@exe/screenshot-area-selection.c.o'.
[1/16] Compiling C object 'src/gnome-screenshot@exe/cheese-flash.c.o'.
[1/16] Compiling C object 'src/gnome-screenshot@exe/gnome-screenshot.c.o'.
[1/16] Compiling C object 'src/gnome-screenshot@exe/screenshot-application.c.o'.
[1/16] Compiling C object 'src/gnome-screenshot@exe/screenshot-config.c.o'.
[2/16] Generating screenshot-resources_c with a custom command.
[2/16] Compiling C object 'src/gnome-screenshot@exe/meson-generated_.._screenshot-resources.c.o'.
[3/16] Compiling C object 'src/gnome-screenshot@exe/screenshot-area-selection.c.o'.
FAILED: src/gnome-screenshot@exe/screenshot-area-selection.c.o 
ccache cc  -Isrc/gnome-screenshot@exe -Isrc -I../src -I. -I../ -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/dbus-1.0/include -I/usr/include/gtk-3.0 -I/usr/include/gio-unix-2.0/ -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/uuid -I/usr/include/freetype2 -I/usr/include/harfbuzz -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c99 -O2 -g -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_DEFAULT_SOURCE -O2 -g -D_REENTRANT -pthread '-DLOCALEDIR="/app/share/locale"' -DGLIB_DISABLE_DEPRECATION_WARNINGS -DGDK_DISABLE_DEPRECATION_WARNINGS -MMD -MQ 'src/gnome-screenshot@exe/screenshot-area-selection.c.o' -MF 'src/gnome-screenshot@exe/screenshot-area-selection.c.o.d' -o 'src/gnome-screenshot@exe/screenshot-area-selection.c.o' -c ../src/screenshot-area-selection.c
ccache: error: /home/builder/.ccache/ccache.conf: No such file or directory
[4/16] Compiling C object 'src/gnome-screenshot@exe/meson-generated_.._screenshot-resources.c.o'.
[5/16] Compiling C object 'src/gnome-screenshot@exe/screenshot-config.c.o'.
[6/16] Compiling C object 'src/gnome-screenshot@exe/gnome-screenshot.c.o'.
[7/16] Compiling C object 'src/gnome-screenshot@exe/cheese-flash.c.o'.
[8/16] Compiling C object 'src/gnome-screenshot@exe/screenshot-application.c.o'.
[9/16] Generating org.gnome.Screenshot.desktop_merge with a custom command.
[10/16] Generating org.gnome.Screenshot.metainfo.xml_merge with a custom command.
ninja: build stopped: subcommand failed.
```
I have no idea why....

**Comment by bilelmoussaoui** [comment]
@TingPing @nedrichards Can you guys trigger a new build to see if it's just a builder issue? :)

**Comment by nedrichards** [comment]
bot, build com.github.bilelmoussaoui.Authenticator

**Comment by bilelmoussaoui** [comment]
It builds successfully now! I have just removed the file system access from the finish args as it doesn't change anything... It's even better to keep the database file sandboxed.

**Comment by nedrichards** [comment]
bot, build com.github.bilelmoussaoui.Authenticator

**Comment by bilelmoussaoui** [comment]
Builds successfully  https://flathub.org/builds/#/builders/2/builds/226 :tada:

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/com.github.bilelmoussaoui.Authenticator

**Pull request closed without merging**